test_runner/src/main/kotlin/ftl/run/NewTestRun.kt

```diff
@@ -23,8 +24,14 @@ suspend fun newTestRun(args: IArgs) = withTimeoutOrNull(args.parsedTimeout) {
     if (!args.async) {
         cancelTestsOnTimeout(args.project, matrixMap.map) { pollMatrices(matrixMap.map.keys, args).updateMatrixMap(matrixMap) }
         cancelTestsOnTimeout(args.project, matrixMap.map) { fetchArtifacts(matrixMap, args) }
-
-        ReportManager.generate(matrixMap, args, testShardChunks, ignoredTests)
+        runCatching {
+            ReportManager.generate(matrixMap, args, testShardChunks, ignoredTests)
```

Comment:
After report generating, flank validates matrices if any in other state than success flank throws an exception, we need to catch this error print matrices, and rethrow the error.  Maybe there is a better way to realize it or maybe we can return the validation state by an enum and throw here. WDYT?

Reply:
Perfect question. IMO, validation is a different step than generating report, so `matrices.validate(args.ignoreFailedTests)` function shouldn't be called in `ReportManager.generate` but in the `newTestRun` and `refreshLastRun`.
After that `runCatching` and rethrow shouldn't be necessary.

Reply:
Thanks for the feedback! I move validation outside of report generation. 👍